src/coreclr/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.CoreCLR.cs

```diff
@@ -19,7 +19,7 @@ public partial class AssemblyLoadContext
 
         [RequiresUnreferencedCode("Types and members the loaded assembly depends on might be removed")]
         [DllImport(RuntimeHelpers.QCall, CharSet = CharSet.Unicode)]
-        private static extern IntPtr LoadFromStream(IntPtr ptrNativeAssemblyLoadContext, IntPtr ptrAssemblyArray, int iAssemblyArrayLen, IntPtr ptrSymbols, int iSymbolArrayLen, ObjectHandleOnStack retAssembly);
+        private static extern void LoadFromStream(IntPtr ptrNativeAssemblyLoadContext, IntPtr ptrAssemblyArray, int iAssemblyArrayLen, IntPtr ptrSymbols, int iSymbolArrayLen, ObjectHandleOnStack retAssembly);
```

Comment:
This looks like a typo that somehow makes no difference. The Qcall returns `void` -

https://github.com/dotnet/runtime/blob/69e114c1abf91241a0eeecf1ecceab4711b8aa62/src/coreclr/vm/assemblynative.cpp#L297

Reply:
> that somehow makes no difference

Interesting.  Does it just end up returning some garbage register value or something?

Reply:
Most likely it was returning in a register that caller immediately used to store something else. 

The original idea was that the mismatch in the returns was the reason for the "leak". Turns out it was a bit deeper than that, and fixing the return type did not change anything.

Reply:
cc: @janvorli

Reply:
The interop method would return whatever was in RAX / EAX / X0 / R0 at that point.